wgsl/index.bs

```diff
@@ -5297,12 +5297,18 @@ See [[#discard-statement]].
     function parameter.
     * In particular, an argument that is a pointer must agree with the formal parameter
         on storage class, pointee type, and access mode.
-* A function parameter of pointer type must be in one of
+* For [=user-defined functions=], a parameter of pointer type must be in one of
     the following storage classes:
     * [=storage classes/function=]
     * [=storage classes/private=]
     * [=storage classes/workgroup=]
-* Each argument of a function call of pointer type must be one of:
+* For [=built-in functions=], a parameter of pointer type must be in one of
+    the following storage classes:
+    * [=storage classes/function=]
+    * [=storage classes/private=]
+    * [=storage classes/workgroup=]
+    * [=storage classes/storage=]
```

Comment:
So a built-in function can accept a pointer into a storage buffer (I assume `arrayLength`), but a user function can't?
Sorry for a dumb question, why am I not allowed to pass a pointer to a matrix from my uniform buffer, to a function?

Reply:
SPIR-V restricts pointers to resources in universal validation (in the non-variable pointers rules); however, in SPIR-V `arrayLength` is not a function, but an instruction. This is the same case for atomics. So WGSL can have builtin functions that are less restrictive because they expose specific hardware functionality that is not generally supported.